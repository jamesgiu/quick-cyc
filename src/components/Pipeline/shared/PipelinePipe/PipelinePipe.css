/* Sizes */
.qc-pipe-wrapper.qc-size-xs {
  height: 5px;
}

.qc-pipe-wrapper.qc-size-s {
  height: 10px;
}

.qc-pipe-wrapper.qc-size-m {
  height: 15px;
}

.qc-pipe-wrapper.qc-size-l {
  height: 20px;
}

.qc-pipe-wrapper.qc-size-xl {
  height: 25px;
}

.qc-pipe-wrapper.qc-state-active {
  background-color: #5f6b7c;
}

.qc-pipe-wrapper {
  overflow: hidden;
  width: 10vw;
  border-radius: 1em;
  transition:
    background-color 0.5s,
    border 0.5s;
}

.qc-pipe {
  font-size: 2em;
  padding: 0.5em 1em;
  box-shadow: rgb(0 0 0 / 24%) 0 3px 8px;

  /* Expanded the size of the image to prevent visual blips when animation loop repeats */
  background-size: 5000%;
}

.qc-pipe-wrapper.qc-state-active .qc-pipe {
  animation: animated-background 200s linear infinite;
}

.qc-pipe-loading-wrapper {
  overflow: hidden;
}

/* Intents */
.qc-pipe-wrapper.qc-state-active.qc-intent-none .qc-pipe {
  background-image: repeating-linear-gradient(
    50deg,
    #5f6b7c,
    #5f6b7c 10px,
    #d2d5da 15px,
    #d2d5da 20px
  );
}

.qc-pipe-wrapper.qc-state-active.qc-intent-in-progress .qc-pipe {
  background-image: repeating-linear-gradient(
    50deg,
    #2d72d2,
    #2d72d2 10px,
    #7ca5dd 15px,
    #7ca5dd 20px
  );
}

.qc-pipe-wrapper.qc-state-active.qc-intent-success .qc-pipe {
  background-image: repeating-linear-gradient(
    50deg,
    #238551,
    #238551 10px,
    #729984 15px,
    #729984 20px
  );
}

.qc-pipe-wrapper.qc-state-active.qc-intent-warning .qc-pipe {
  background-image: repeating-linear-gradient(
    50deg,
    #fbb360,
    #fbb360 10px,
    #e6caab 15px,
    #e6caab 20px
  );
}

.qc-pipe-wrapper.qc-state-active.qc-intent-failure .qc-pipe {
  background-image: repeating-linear-gradient(
    50deg,
    #cd4246,
    #cd4246 10px,
    #c98588 15px,
    #c98588 20px
  );
}

.qc-pipe-wrapper.qc-state-active.qc-intent-skipped .qc-pipe {
  background-image: repeating-linear-gradient(
    50deg,
    #5f6b7c,
    #5f6b7c 10px,
    #d2d5da 15px,
    #d2d5da 20px
  );
  opacity: 0.1;
}

@keyframes animated-background {
  from {
    background-position: 0 0;
  }

  to {
    background-position: -100% 0;
  }
}
