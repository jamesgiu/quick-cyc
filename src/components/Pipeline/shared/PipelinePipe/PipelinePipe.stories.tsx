import type { Meta, StoryObj } from '@storybook/react';

import { PipelinePipe } from './PipelinePipe';
import { PipelineIntent, PipelineSize } from '../../types';

const meta: Meta<typeof PipelinePipe> = {
    title: 'Pipe',
    component: PipelinePipe,
    parameters: {
      // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
      layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs']
  };

type Story = StoryObj<typeof PipelinePipe>;
  
export const Primary: Story = {
    args: {
      intent: PipelineIntent.IN_PROGRESS,
      active: true,
      size: PipelineSize.M,
    }
};
  
  
export const Progress: Story = {
  args: {
    intent: PipelineIntent.IN_PROGRESS,
    active: true,
    size: PipelineSize.M,
    progressPercent: 30,
  }
};

      
export default meta;