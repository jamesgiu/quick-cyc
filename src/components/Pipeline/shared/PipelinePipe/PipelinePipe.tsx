import React from "react";
import { PipelineIntent, PipelineSize } from "../../types";
import "./PipelinePipe.css";

export interface PipelinePipeProps {
    intent?: PipelineIntent,
    active?: boolean,
    size: PipelineSize,
    progressPercent?: number,
}

export const PipelinePipe = (props: PipelinePipeProps) => {
    return (
        <div className={"qc-pipe-wrapper" + ` ${props.size} ${props.intent} ${props.active ? "qc-state-active" : ""}`}>
            <div className={"qc-pipe-loading-wrapper"} style={{width: props.progressPercent + "%"}}>
                <div className={"qc-pipe"}/>
            </div>
        </div>
    );
  };
  
  export default PipelinePipe;