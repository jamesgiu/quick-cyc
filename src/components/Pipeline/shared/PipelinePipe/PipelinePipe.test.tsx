// @vitest-environment jsdom

import { expect, test } from "vitest";

import { composeStories } from "@storybook/react-vite";

import * as stories from "./PipelinePipe.stories";

const { Primary } = composeStories(stories);
test("Snapshot test", async () => {
  await Primary.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
