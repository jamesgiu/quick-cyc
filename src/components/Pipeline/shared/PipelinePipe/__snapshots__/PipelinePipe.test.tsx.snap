// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot test 1`] = `
<div>
  <div
    class="qc-pipe-wrapper qc-size-m qc-intent-in-progress qc-state-active"
  >
    <div
      class="qc-pipe-loading-wrapper"
    >
      <div
        class="qc-pipe"
      />
    </div>
  </div>
</div>
`;
